releasenotes/notes/remove-symbolic-pulse-subclasses-77314a1654521852.yaml

```diff
@@ -0,0 +1,15 @@
+---
+upgrade:
+  - |
+    Symbolic pulse subclasses :class:`.Gaussian`, :class:`.GaussianSquare`,
+    :class:`.Drag` and :class:`.Constant` have been upgraded to
+    instantiate :class:`SymbolicPulse` rather than the subclass itself.
+    All parametric pulse object in pulse programs must be symbolic pulse instance,
+    because subclassing is no longer neccesary. Note that :class:`SymbolicPulse` can
+    uniquely identify a particular envelope with the symbolic expression object
+    defined in :attr:`SymbolicPulse.envelope`.
+  - |
+    `isinstance` check with pulse classes :class:`.Gaussian`, :class:`.GaussianSquare`,
+    :class:`.Drag` and :class:`.Constant` has been invalidated because
+    these classes are never instantiated.
+    Instead of using type information, :attr:`SymbolicPulse.pulse_type` must be used.
```

Comment:
If `pulse_type` was already in use as an alias, it might be a little safer to do the instance check on some private internal string instead, just to avoid cases where the user overrides the alias and still wants the type check to work.

Reply:
I wanted to remove the usage of subclass since this will never work properly with custom pulse defined by user or third party library, as described here: https://github.com/Qiskit/qiskit-terra/pull/8278#discussion_r968580314